app/src/main/java/com/example/grocerieswizard/ui/home/HomeFragment.java

```diff
@@ -17,41 +16,40 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import com.example.grocerieswizard.R;
-import com.example.grocerieswizard.data.local.model.CartItem;
-import com.example.grocerieswizard.ui.addrecipe.AddRecipeFragment;
-import com.example.grocerieswizard.data.repo.RecipeRepository;
 import com.example.grocerieswizard.databinding.FragmentHomeBinding;
-import com.example.grocerieswizard.ui.detail.DetailFragment;
 import com.example.grocerieswizard.di.GroceriesWizardInjector;
+import com.example.grocerieswizard.ui.UiMapper;
+import com.example.grocerieswizard.ui.addrecipe.AddRecipeFragment;
+import com.example.grocerieswizard.ui.detail.DetailFragment;
 import com.example.grocerieswizard.ui.model.IngredientUi;
 import com.example.grocerieswizard.ui.model.RecipeUi;
-import com.example.grocerieswizard.ui.UiMapper;
 
 import java.util.List;
-import java.util.stream.Collectors;
-
-public class HomeFragment extends Fragment implements RecipeInterface {
 
+public class HomeFragment extends Fragment implements RecipeInterface, HomeContract.View {
+    HomeContract.Presenter presenter;
     private RecipeRecyclerViewAdapter adapter;
-    private RecipeRepository recipeRepository;
-    private UiMapper uiMapper;
-    private static final String TAG = "HomeFragment";
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         adapter = new RecipeRecyclerViewAdapter();
         adapter.setRecyclerViewInterface(this);
         GroceriesWizardInjector injector = GroceriesWizardInjector.getInstance();
-        recipeRepository = injector.getRecipeRepository();
-        uiMapper = injector.getUiMapper();
+        UiMapper uiMapper = injector.getUiMapper();
+        presenter = new HomePresenter(injector.getRecipeRepository(), uiMapper);
     }
 
-    @NonNull
-    private List<RecipeUi> getAllRecipes() {
-        return recipeRepository.getAllRecipes().stream()
-                .map(uiMapper::toRecipeUi)
-                .collect(Collectors.toList());
+    @Override
+    public void onStart() {
+        super.onStart();
+        presenter.bindView(this);
+    }
+
+    @Override
+    public void onStop() {
+        presenter.unbindView();
```

Comment:
Just want to mention something here: This part is called cleaning resources, and whenever you clear resources you first clear them and then call the super method. And when you are creating resources you first call the super method and the clear the resources.